### PR TITLE
make cfile root and file blocks dynamic

### PR DESCRIPTION
**Pull request by notimaginative**
Remove limit on number of roots and files supported by cfilesystem.

### CODE REVIEW

**Comment by notimaginative** [comment]
Works on my limited data set but needs further testing with very large numbers of roots/files.

File: code/cfile/cfilesystem.cpp [hunk 1]
```diff
@@ -72,62 +72,65 @@ typedef struct cf_root_sort {
...
+#define CF_NUM_FILES_PER_BLOCK		512
+#define CF_MIN_FILE_BLOCKS			16				// minimum number of file blocks to create on resize
 
 typedef struct cf_file_block {
 	cf_file						files[CF_NUM_FILES_PER_BLOCK];
 } cf_file_block;
 
 static uint Num_files = 0;
-static cf_file_block  *File_blocks[CF_MAX_FILE_BLOCKS];
+static SCP_vector<cf_file_block *> File_blocks;
```
Comment (asarium) [inline code/cfile/cfilesystem.cpp:109]: You could use `unique_ptr`s to avoid the manual memory management here.

**Review by asarium — APPROVED** [review]

File: code/cfile/cfilesystem.cpp (hunk 1 above)
Comment (notimaginative) [reply code/cfile/cfilesystem.cpp]: Good point. I don't really use them enough for that to be my goto for stuff like this.

**Comment by EatThePath** [comment]
> Works on my limited data set but needs further testing with very large numbers of roots/files.

@notimaginative how large is usefully large for testing?

**Comment by notimaginative** [comment]
Preferably above 70k, which puts it over the old limit. But anything over 50k would probably be a good enough high mark to verify everything works.

**Comment by notimaginative** [comment]
Tested with over 100k files without issue so this should be as ready as it's going to get.

**Pull request merged**